src/components/signUpInOut/SignUpForm.jsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, Row, Col, Modal,
+} from 'react-bootstrap';
+import { FaXmark } from 'react-icons/fa6';
+import { signUp } from '../../features/authSlice';
+
+const initialAddressData = {
+  street: '',
+  city: '',
+  state: '',
+  zip_code: '',
+};
+
+const initialFormData = {
+  name: '',
+  age: '',
+  email: '',
+  photo: '',
+  role: '',
+  password: '',
+  password_confirmation: '',
+  address: { ...initialAddressData },
+};
+
+const SignUpForm = ({ showSignUpModal, handleCloseSignUpModal, handleOpenSignUpModal }) => {
+  const [formData, setFormData] = useState({ ...initialFormData });
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevFormData) => {
+      if (name.startsWith('address.')) {
+        const addressField = name.split('.')[1];
+        return {
+          ...prevFormData,
+          address: {
+            ...prevFormData.address,
+            [addressField]: value,
+          },
+        };
+      }
+      return {
+        ...prevFormData,
+        [name]: value,
+      };
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(signUp(formData));
+    window.location.href = '/';
+  };
+
+  return (
```

Comment:
- [ ] When I add a doctor, the photo can overflow the content:

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/9ac984f9-5c90-495e-83e3-82320b5dbfe9)

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/b9044850-27c4-4ac2-9f64-4ff0a872eb42)

Kindly set limits on your CSS for the doctor's photos.